linode/instancenetworking/framework_datasource.go

```diff
@@ -0,0 +1,82 @@
+package instancenetworking
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{}
+}
+
+type DataSource struct {
+	client *linodego.Client
+}
+
+func (d *DataSource) Configure(
+	ctx context.Context,
+	req datasource.ConfigureRequest,
+	resp *datasource.ConfigureResponse,
+) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	meta := helper.GetDataSourceMeta(req, resp)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	d.client = meta.Client
+}
+
+func (d *DataSource) Metadata(
+	ctx context.Context,
+	req datasource.MetadataRequest,
+	resp *datasource.MetadataResponse,
+) {
+	resp.TypeName = "linode_instance_networking"
+}
+
+func (d *DataSource) Schema(
+	ctx context.Context,
+	req datasource.SchemaRequest,
+	resp *datasource.SchemaResponse,
+) {
+	resp.Schema = frameworkDatasourceSchema
+}
+
+func (d *DataSource) Read(
+	ctx context.Context,
+	req datasource.ReadRequest,
+	resp *datasource.ReadResponse,
+) {
+	client := d.client
+
+	var data DataSourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	netInfo, err := client.GetInstanceIPAddresses(ctx, int(data.LinodeID.ValueInt64()))
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to get Instance Networking Information: ", err.Error(),
+		)
+		return
+	}
+
+	data.parseInstanceIPAddressResponse(ctx, netInfo)
+
+	resp.Diagnostics.Append(data.parseInstanceIPAddressResponse(ctx, netInfo)...)
```

Comment:
```suggestion
	resp.Diagnostics.Append(data.parseInstanceIPAddressResponse(ctx, netInfo)...)
```

I think you would only need to call `data.parseInstanceIPAddressResponse` once. Or is there any reason to call it twice?

Reply:
You're right. Just a copy-paste mistake. I will fix this!